src/phpDebug.ts

```diff
@@ -458,6 +463,12 @@ class PhpDebugSession extends vscode.DebugSession {
                 // for all connections
                 await Promise.all(
                     connections.map(async (connection, connectionIndex) => {
+                        if (hasWaitingConnections && connection.isPendingExecuteCommand()) {
+                            // skip this if there is a connection that is being initialized, and this
+                            // connection is in middle of running PHP code. This avoids a deadlock that
+                            // can happen if the PHP code initializes a new connection that uses xdebug
+                            return;
```

Comment:
Simply ignoring this setBreakpoint command seems brittle. The way I would imagine it is to still send the command (queue it) but don't wait for the response (and send a potential error to the debug console).

Reply:
I'll see if I can get it to work like that.  I was hesitant to try that way as I think it would involve the exact same code (so a lot of duplicated code), just without the `await`, instead maybe using `.then()`.

I'll try it and see if it works like that.